wfe/wfe_test.go

```diff
@@ -942,6 +942,27 @@ func TestChallenge(t *testing.T) {
 	test.AssertEquals(t, responseWriter.Code, http.StatusNotFound)
 	assertJSONEquals(t, responseWriter.Body.String(),
 		`{"type":"urn:acme:error:malformed","detail":"Expired authorization","status":404}`)
+
+	//Challenge Not found
+	challengeURL = "not_found/23"
+	responseWriter = httptest.NewRecorder()
+	wfe.Challenge(ctx, newRequestEvent(), responseWriter,
+		makePostRequestWithPath(challengeURL,
+			signRequest(t, `{"resource":"challenge"}`, wfe.nonceService)))
+	test.AssertEquals(t, responseWriter.Code, http.StatusNotFound)
+	assertJSONEquals(t, responseWriter.Body.String(),
+		`{"type":"urn:acme:error:malformed","detail":"No such challenge","status":404}`)
+
+	// When something goes wrong
+	errorUrl := "error/24"
```

Comment:
I'm not positive that this is testing the codepath that you think it is, e.g. the one where a err is returned from the `SelectOne` in `GetAuthorization()` other than `sql.ErrNoRows`. Am I incorrect about that? Could you perhaps flesh out the "// When something goes wrong" comment with more detail if so?

Thanks!